language/tools/vm-genesis/src/lib.rs

```diff
@@ -396,16 +401,30 @@ fn create_and_initialize_owners_operators(
     let libra_root_address = account_config::libra_root_address();
 
     // Create accounts for each validator owner
-    for (owner_key, owner_name, _) in operator_assignments {
-        let owner_auth_key = AuthenticationKey::ed25519(&owner_key);
-        let owner_account = account_address::from_public_key(owner_key);
+    for (owner_key, owner_name, _op_assignment) in operator_assignments {
```

Comment:
I think restructuring this might make it a little clearer as to why we need to do the hack of rotate_authentication_key() after we've just created the account. From what I understand, we only have to rotate in the case that we don't have a valid auth key, because the create_validator_account() call will fail with the invalid auth key, right? If so, can we make this more obvious? Right now it's not super clear.. I guess the issue is that we need to use a dummy auth key on the creation, first? Perhaps adding a little more duplication will make it clearer where the hack is? e.g.,

```rust
let owner_address = libra_config::utils::validator_owner_account_from_name(owner_name);
if let Some(owner_key) = owner_key {
    let auth_key = AuthenticationKey::ed25519(owner_key).to_vec();
    // Create owner script and call exec()
} else {
    let dummy_auth_key = AuthenticationKey::random();
    // Create owner script and call exec()
    // Rotate using the ZERO_AUTH_KEY
}
```

If you'd prefer to keep your structure, you could just add a comment as to what's going on here. 

Also, does it make sense to use the dummy_auth_key instead of staged_owner_auth_key anyway? This will help to keep the default_validator_owner_auth_key_from_name(..) method private?

Reply:
that doesn't work.

we only pass in a portion of the auth key and the other portion is the address... so the initial authkey is really useless..

Reply:
I guess I'll try to clear up some of the confusion around account creation in move here, even though this is not the best place.

Reply:
When you say "that doesn't work", what are you referring to? Given that the initial authkey is useless, at the very least you could generate a random one, right? So that way there's no need to make "default_validator_owner_auth_key_from_name" public and that'll be a little bit cleaner?

Don't feel too pressured to fix the move side. My comment was more about providing context/an explanation to the reader about why the hack exists, rather than fixing the hack

Reply:
yeah, I gave a comment blurb up above now. You do make an interesting point.... but then we need to have something that generates half an auth key... hacks hack everywhere hacks